content/docs/optimizing-performance.md

```diff
@@ -185,13 +185,31 @@ Currently Chrome, Edge, and IE are the only browsers supporting this feature, bu
 If your application renders long lists of data (hundreds or thousands of rows), we recommended using a technique known as "windowing". This technique only renders a small subset of your rows at any given time, and can dramatically reduce the time it takes to re-render the components as well as the number of DOM nodes created.
 
 [React Virtualized](https://bvaughn.github.io/react-virtualized/) is one popular windowing library. It provides several reusable components for displaying lists, grids, and tabular data. You can also create your own windowing component, like [Twitter did](https://medium.com/@paularmstrong/twitter-lite-and-high-performance-react-progressive-web-apps-at-scale-d28a00e780a3), if you want something more tailored to your application's specific use case.
- 
+
 ## Avoid Reconciliation
 
 React builds and maintains an internal representation of the rendered UI. It includes the React elements you return from your components. This representation lets React avoid creating DOM nodes and accessing existing ones beyond necessity, as that can be slower than operations on JavaScript objects. Sometimes it is referred to as a "virtual DOM", but it works the same way on React Native.
 
 When a component's props or state change, React decides whether an actual DOM update is necessary by comparing the newly returned element with the previously rendered one. When they are not equal, React will update the DOM.
 
+You can now visualize these re-renders of the virtual DOM with React DevTools:
+
+- [Chrome Browser Extension](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)
+- [Firefox Browser Extension](https://addons.mozilla.org/en-GB/firefox/addon/react-devtools/)
+- [Standalone Node Package](https://www.npmjs.com/package/react-devtools)
+
+In the developer console select the **Highlight Updates** option in the **React** tab
+
+<img src="https://i.imgur.com/tUBa5k7.png" style="max-width:100%; margin-top:10px;" alt="How to enable highlight updates" />
+
+Interact with your page and you should see colored borders momentarily appear around any elements that have redrawn.
+
+You should expect to correctly see redraws on components that are updating in the DOM, but you shouldn't see them around components that are not updating.  So if you see flashing colors around components that should not be updating then you know you have a needless `render()`.
+
+Below is an example of an [app](https://highlight-demo.firebaseapp.com/) setup specifically to demonstrate a very poorly optimized redraw, everything is being re-rendered in the virtual DOM even though only the text input component needs to be re-rendered.  More info can be found in this [blog post](https://blog.logrocket.com/make-react-fast-again-part-3-highlighting-component-updates-6119e45e6833) from [Ben Edelstein](https://blog.logrocket.com/@edelstein).
```

Comment:
We try to avoid "virtual DOM" in the documentation except one or two cases. Let's say:

>Below is an example of an app setup specifically to demonstrate a poorly optimized redraw. Even though only the text input changes, the components outside it are being re-rendered. Even though React only updates the changed DOM nodes, re-rendering still takes some time. In many cases it's not a problem, but if the slowdown is noticeable, we can fix it by avoiding reconciliation.